skpro/distributions/histogram.py

```diff
@@ -0,0 +1,73 @@
+# copyright: skpro developers, BSD-3-Clause License (see LICENSE file)
+"""Histogram distribution."""
+
+__author__ = ["ShreeshaM07"]
+
+import numpy as np
+
+from skpro.distributions.base import BaseDistribution
+
+# import pandas as pd
+
+
+class Histogram(BaseDistribution):
+    """Histogram Probability Distribution.
+
+    The histogram probability distribution is parameterized
+    by the bins and bin densities.
+
+    Parameters
+    ----------
+    bins : float or array of float 1D
+        array has the bin boundaries with 1st element the first bin's
+        starting point and rest are the bin ending points of all bins
+    bin_density: array of float 1D
+        The density of bins.
+        i.e., it is equal to the empirical probability divided
+        by the interval length, or bin width.
+    index : pd.Index, optional, default = RangeIndex
+    columns : pd.Index, optional, default = RangeIndex
+    """
+
+    def __init__(self, bins, bin_density, index=None, columns=None):
+        self.bins = bins
+        self.bin_density = bin_density
+
+        super().__init__(index=index, columns=columns)
+
+    def _pdf(self, x):
+        """Probability density function.
+
+        Parameters
+        ----------
+        x : 1D np.ndarray, same shape as ``self``
+            values to evaluate the pdf at
+
+        Returns
+        -------
+        1D np.ndarray, same shape as ``self``
+            pdf values at the given points
+        """
+        bin_density = np.array(self.bin_density.copy())
+        bins = self.bins
+        pdf = []
+        if isinstance(bins, list):
```

Comment:
main comment, this looks correct, but it is quite inefficient due to the use of loops.

I would strongly advise to use `numpy` methods for everything.

For instance, for bin widths, use `diff`.

To find the bin in which the x-value falls, you could use `cumsum` and `np.where` with `>`.